master/internal/api_experiment.go

```diff
@@ -1390,3 +1394,47 @@ func (a *apiServer) MoveExperiment(
 	return &apiv1.MoveExperimentResponse{},
 		errors.Wrapf(err, "error moving experiment (%d)", req.ExperimentId)
 }
+
+func (a *apiServer) GetModelDefTree(
+	_ context.Context, req *apiv1.GetModelDefTreeRequest,
+) (*apiv1.GetModelDefTreeResponse, error) {
+	tgz, err := a.m.db.ExperimentModelDefinitionRaw(int(req.ExperimentId))
+	if err != nil {
+		return nil, errors.Wrapf(err,
+			"error fetching model definition from database: %d", req.ExperimentId)
+	}
+	var resp apiv1.GetModelDefTreeResponse
+	arc, err := archive.FromTarGz(tgz)
```

Comment:
To me, this suggests that we need to consider _not_ storing model definitions in the database anymore.

This is going to be pretty expensive: we're going to query up to 100MB from the database and gunzip all of it in order to get just a list of files in the checkpoint?

The GetModelDefFile has similar shortcomings, it's going to query/gunzip up to 100MB of data to fetch a single file.

And I'd bet that somewhere, somebody is going to try to call GetModelDefFile once for every file in the GetModelDefTree, which is going to query/gunzip up to 100MB of data once for every single file in the model definition.